harness/harness-core-ui
# fix: [PIE-2805]: support added for runtime input for when condition

##### Summary:



##### Jira Links:

https://harness.atlassian.net/browse/PIE-2805

##### Screenshots:




https://user-images.githubusercontent.com/96409598/155131717-029f7c2e-3968-40a0-b4dd-2b3a671d219a.mov



https://user-images.githubusercontent.com/96409598/155131825-2c8e235e-b3e8-40ff-ad2c-8f114c6b9777.mov


https://user-images.githubusercontent.com/96409598/156038141-3f3a28be-28f0-4ef1-800e-1327e4670f30.mov



**Stage skipped for JEXL condition resulting to false -->** 


https://user-images.githubusercontent.com/96409598/156458710-7198283d-8caf-4e69-8866-794be5651d2a.mov

<img width="1792" alt="Screenshot 2022-03-03 at 3 43 14 AM" src="https://user-images.githubusercontent.com/96409598/156458841-a7ee8252-829d-46d3-bcad-d2a84653c096.png">


**Stage executed for JEXL condition resulting to true -->** 



https://user-images.githubusercontent.com/96409598/156459402-d12e669c-71ac-4b89-9c8c-e0084e546c74.mov


After the latest set of changes --> 



https://user-images.githubusercontent.com/96409598/156558303-12a73688-0b50-4f50-a7ee-4a6831b0724b.mov



<img width="1792" alt="Screenshot 2022-03-03 at 5 02 37 PM" src="https://user-images.githubusercontent.com/96409598/156558341-dac3bf57-84bf-4d2a-a179-1504a55af9c7.png">


<img width="1792" alt="Screenshot 2022-03-03 at 5 02 55 PM" src="https://user-images.githubusercontent.com/96409598/156558359-f062ef32-7308-46c2-a9d8-74ff4b1ac476.png">


<img width="1792" alt="Screenshot 2022-03-03 at 5 08 25 PM" src="https://user-images.githubusercontent.com/96409598/156558370-02b5cc45-63cb-4cf0-add9-53f02ae39da5.png">



**All skipped (Stage + step) Testing -------**


https://user-images.githubusercontent.com/96409598/156584551-9a66dd78-31af-455f-a9ee-479afe327c75.mov



**All true (Stage + step) Testing -------**



https://user-images.githubusercontent.com/96409598/156584643-89cdd066-fac4-4bbb-9a5a-8039b65f0444.mov



Runtime component added -----


<img width="1792" alt="Screenshot 2022-03-04 at 1 08 24 AM" src="https://user-images.githubusercontent.com/96409598/156640210-f5e0e22b-b043-4f2a-9901-939d8cf23dfb.png">




You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`
